# Used Cars Sales Analysis

## Task 1 

In [1]:
import pandas as pd
import numpy as np

In [2]:
 price = [80121, 48483, 37077, 30660, 68536, 52666, 78703, 51036, 55643, 56990, 39967, 93822, 51406, 93947, 78373, 62765, 47536, 40528, 58780, 83197] 

In [3]:
Data = pd.Series(price, index=np.arange(101,121))

In [4]:
Data

101    80121
102    48483
103    37077
104    30660
105    68536
106    52666
107    78703
108    51036
109    55643
110    56990
111    39967
112    93822
113    51406
114    93947
115    78373
116    62765
117    47536
118    40528
119    58780
120    83197
dtype: int64

Conclusion - Created a pandas series for the prices of cars with index from 101 to 120 

## Task 2

In [5]:
data=pd.DataFrame()

In [6]:
data["Index"]=np.arange(101,121)
data["price"]=price

In [7]:
data

Index  price
0     101  80121
1     102  48483
2     103  37077
3     104  30660
4     105  68536
5     106  52666
6     107  78703
7     108  51036
8     109  55643
9     110  56990
10    111  39967
11    112  93822
12    113  51406
13    114  93947
14    115  78373
15    116  62765
16    117  47536
17    118  40528
18    119  58780
19    120  83197

In [8]:
data_sort = data.sort_values(by ='price')

In [9]:
data_sort

Index  price
3     104  30660
2     103  37077
10    111  39967
17    118  40528
16    117  47536
1     102  48483
7     108  51036
12    113  51406
5     106  52666
8     109  55643
9     110  56990
18    119  58780
15    116  62765
4     105  68536
14    115  78373
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

In [10]:
data_sort.iloc[0]

Index      104
price    30660
Name: 3, dtype: int64

In [11]:
data_sort.iloc[19]

Index      114
price    93947
Name: 13, dtype: int64

Conclusion - The details for the highest and lowest prices of cars is found.

## Task 3 

In [12]:
data_sort.head(5)     #head displays the top values

Index  price
3     104  30660
2     103  37077
10    111  39967
17    118  40528
16    117  47536

In [13]:
data_sort.tail(5)     #Tail displays the top values

Index  price
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

Conclusion - The price of the top 5 cars with highest value and lowest value found 

## Task 4 

In [14]:
betn_30_and_40k =data_sort[(data_sort['price']>30000) & (data_sort['price']<40000)]

In [15]:
betn_30_and_40k

Index  price
3     104  30660
2     103  37077
10    111  39967

In [16]:
len(betn_30_and_40k)

3

Conclusion - The prices of cars that are priced between 30k and 40k are found.The the count of it is 3

## Task 5 

In [17]:
avg=data_sort["price"].mean()

In [18]:
avg

60511.8

In [19]:
above_avg=data_sort[(data_sort['price']>60511.8)]

In [20]:
above_avg

Index  price
15    116  62765
4     105  68536
14    115  78373
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

In [36]:
len(above_avg)

8

Conclusion - These are the cars which are priced above the average prices of all cars.The number of cars is found to be 8

# Task 6

In [21]:
add=[34000,45000,54000]

In [22]:
new=pd.DataFrame()

In [23]:
new["Index"]=np.arange(201,204)
new["price"]=add

In [24]:
new

Index  price
0    201  34000
1    202  45000
2    203  54000

In [25]:
updated_data = pd.concat([data, new],ignore_index=True)

In [26]:
updated_data

Index  price
0     101  80121
1     102  48483
2     103  37077
3     104  30660
4     105  68536
5     106  52666
6     107  78703
7     108  51036
8     109  55643
9     110  56990
10    111  39967
11    112  93822
12    113  51406
13    114  93947
14    115  78373
15    116  62765
16    117  47536
17    118  40528
18    119  58780
19    120  83197
20    201  34000
21    202  45000
22    203  54000

Conclusion - Three new cars are being added to the cars with row indexes 201,202 and 203 

# Task 7

In [27]:
above_60000=data_sort[(data_sort['price']>60000)]

In [28]:
above_60000

Index  price
15    116  62765
4     105  68536
14    115  78373
6     107  78703
0     101  80121
19    120  83197
11    112  93822
13    114  93947

In [29]:
disc=pd.DataFrame(above_60000['price']*0.8)

In [30]:
disc

price
15  50212.0
4   54828.8
14  62698.4
6   62962.4
0   64096.8
19  66557.6
11  75057.6
13  75157.6

Conclusion - These are the 20% discounted price of the cars whose prices are greaater than $ 60,000

## Task 8 

In [31]:
len(updated_data)

23

In [32]:
indexes_to_delete = [15, 4, 14, 6, 0, 19, 11, 13]

In [33]:
updated_data = updated_data.drop(indexes_to_delete)

In [34]:
updated_data

Index  price
1     102  48483
2     103  37077
3     104  30660
5     106  52666
7     108  51036
8     109  55643
9     110  56990
10    111  39967
12    113  51406
16    117  47536
17    118  40528
18    119  58780
20    201  34000
21    202  45000
22    203  54000

In [35]:
len(updated_data)

15

Conclusion - Before deleting there are 23 car data and after deleting there are only 15 car data. 